### ISSUE TITLE
TCR option to do "quick commit" when tests pass

### ISSUE THREAD

**Issue by dkandalov** [dkandalov/limited-wip#15]

**Comment by keraton** [comment]
Hi, I prefer to be able automatic commit when tests pass, no need to do anything

**Comment by dkandalov** [comment]
Yes, this is what "quick commit" is about :)

**Comment by keraton** [comment]
Oh ok, I look a bit the code, now I understand what you mean.
Here tested some modification in your code 

```
fun openCommitDialog(changes: List<Change>? = null) {
    ApplicationManager.getApplication().invokeLater({
        val actionEvent = anActionEvent(changes)
        QuickCommitAction().actionPerformed(actionEvent) // Use the QuickCommit ☺️
    }, ModalityState.NON_MODAL)
}
```

And it works nicely except for the commit message. => 0, 1, 
So what is your plan ? Maybe I can help you with this.

**Comment by dkandalov** [comment]
No exact plan but I was going to give a try tomorrow.

Yes, this is the problem with QuickCommit it takes last message and adds a number to it. Maybe it's worth dropping this functionality and just use the last message.

But then the question is how to specify any meaningful commit message when in TCR mode 🤔

**Comment by keraton** [comment]
Hi I think you can do it like this : 

- if the user select the "Open commit dialog on passed test" but if the user doesn't select it, the it should be the quick commit. Actually with this configuration there is nothing happened.
- About the commit message, I'm not sure that the plugin manage to pick up the last message, I only have empty message + number, I don't know why.

**Comment by dkandalov** [comment]
Sorry it took me a while to look at the issue.
It's strange that you don't get message when doing quick commit. I cannot reproduce it. 
Which IDE and version control system are you using?

For now I replaced "Open commit dialog on passed test" with "On passed test" dropdown with the following options:
- open commit dialog
- commit
- commit and push

I also made quick commit use the last message as it is (without numbers). Might make it configurable some time later to be "last commit", "change list name" or "current task name".
